src/convertToObject.js

```diff
@@ -6,7 +6,24 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  if (typeof sourceString !== 'string' || sourceString.trim() === '') {
+    return {};
+  }
+
+  return sourceString
+    .split(';')
+    .map((rule) => rule.trim())
+    .filter((rule) => rule)
+    .reduce((acc, rule) => {
+      const [property, ...valueParts] = rule.split(':');
+      const value = valueParts.join(':').trim();
+
+      if (property && value) {
```

Comment:
Consider adding a check to ensure that both `property` and `value` are not empty after trimming. This will prevent adding entries with empty keys or values to the resulting object.